### PR TITLE
lp1955159: Prevent resetting of played history in database

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1955159

### CODE REVIEW

**Review by Holzhaus — APPROVED** [review]
Code looks good, I leave it up to @ywwg to confirm that it actually fixes the bug.

**Comment by ywwg** [comment]
I don't have a solid way to reproduce the bug yet.  I think this code looks good and is ok to merge now, and then I can see if I can reproduce the issue later.

**Review by ywwg — APPROVED** [review]

**Pull request merged**
